File: Agile Development Practices/Code/DotNet/videoworld/VideoWorld/Models/ChildrensPrice.cs
using System;

namespace VideoWorld.Models
{
    public class ChildrensPrice : IPrice
    {
        public decimal GetCharge(int daysRented)
        {
            decimal result = 1.50m;
            if (daysRented > 3)
                result += (daysRented - 3) * 1.50m;
            return result;
        }

        public int GetFrequentRenterPoints(int daysRented)
        {
            return 1;
        }
    }
}

File: Agile Development Practices/Code/Java/videoworld/src/main/java/com/thoughtworks/videorental/domain/ChildrensPrice.java
package com.thoughtworks.videorental.domain;

public class ChildrensPrice implements Price {

	public double getCharge(final int daysRented) {
		double result = 1.5;
		if (daysRented > 3)
			result += (daysRented - 3) * 1.5;
		return result;
	}

	public int getFrequentRenterPoints(final int daysRented) {
		return 1;
	}

}
